src/core/execution_stage.cc

```diff
@@ -67,6 +67,20 @@ execution_stage* execution_stage_manager::get_stage(const sstring& name) {
     return _stages_by_name[name];
 }
 
+void execution_stage_manager::update_scheduling_group_name(const sstring& new_sg_name) {
+    for (auto* execution_stage_ptr : _execution_stages) {
+        if (execution_stage_ptr != nullptr && execution_stage_ptr->get_sg().name() == new_sg_name) {
+            auto old_name = execution_stage_ptr->name();
+            execution_stage_ptr->rename();
+            _stages_by_name.erase(old_name);
+            auto ret = _stages_by_name.emplace(execution_stage_ptr->name(), execution_stage_ptr);
+            if (!ret.second) {
+                throw std::invalid_argument(format("Execution stage {} already exists.", execution_stage_ptr->name()));
+            }
+        }
```

Comment:
should we stop after finding stage with new_sg_name ? is it many to many relation or one to one?